src/strcasecmp.c

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <errno.h>
+#include <string.h>
+
+#include "rcutils/strcasecmp.h"
+
+int
+rcutils_strcasecmp(const char * s1, const char * s2)
+{
+  RCUTILS_CAN_FAIL_WITH({errno = EINVAL; return -1;});
+
+  if (s1 == NULL || s2 == NULL) {
+    errno = EINVAL;
+    return -1;
```

Comment:
I think we need to comment `errno` for this API, because the result '-1' might be a case that "s1" is less than "s2".

- The test case below expecting '-1' might be not good.

- About the test case, do you think we need to expect the return value in the cases is (-1, 0, 1), rather than 'NE 0'.

Reply:
Or maybe using LT GT is better.

Reply:
Yeah, this is definitely a problem.  While there are some APIs that rely on errno for this (like `strtod`), it is hard to use them properly.  My suggestion here is to change it so that the return value is either 0 for success or -1 for failure.  Then we add in a pointer out argument which is the comparison value.  That way it is easy for the caller to tell whether it failed or not, and we disambiguate.

Reply:
@clalancette 

that makes sense to me.